### PR TITLE
Fix: correct some mistaken Romanian real towns

### PR DESCRIPTION
**Pull request by ALEX11BR**
Galași -> Galați
Brațov -> Brașov
Băilesti -> Băilești
Călărasi -> Călărași
Drăgăsani -> Drăgășani
Făgăras -> Făgăraș
Târgoviste -> Târgoviște
Târgu Mures ->Târgu Mureș
Tăndărei -> Țăndărei

## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')
* This PR touches english.txt or translations? Check the [guidelines](https://github.com/OpenTTD/OpenTTD/blob/master/docs/eints.md)
* This PR affects the save game format? (label 'savegame upgrade')
* This PR affects the GS/AI API? (label 'needs review: Script API')
    * ai_changelog.hpp, gs_changelog.hpp need updating.
    * The compatibility wrappers (compat_*.nut) need updating.
* This PR affects the NewGRF API? (label 'needs review: NewGRF')
    * newgrf_debug_data.h may need updating.
    * [PR must be added to API tracker](https://wiki.openttd.org/en/Development/NewGRF/Specification%20Status)

### CODE REVIEW

File: src/table/townname.h [hunk 1]
```diff
@@ -2015,15 +2015,15 @@ static const char * const _name_romanian_real[] = {
...
+	u8"T\u00e2rgovi\u0219te",
 	u8"T\u00e2rgu Jiu",
-	u8"T\u00e2rgu Mures",
+	u8"T\u00e2rgu Mure\u0219",
 	"Tecuci",
 	u8"Timi\u0219oara",
 	"Tulcea",
 	"Turda",
 	u8"Turnu M\u0103gurele",
+	u8"\u021a\u0103nd\u0103rei",
```
Comment (nielsmh) [inline src/table/townname.h]: You should not change the order of the names, even if that will make them sorted wrong in the list. If you change the order, existing saved games will suddenly have towns change name and confuse the player.

**Review by nielsmh — DISMISSED** [review]

**Review by nielsmh — COMMENTED** [review]
The commit checker complains about your commit message, you can't have non-ASCII characters in it.
The actual change looks fine to me now.

**Review by LordAro — APPROVED** [review]
Assuming there's no issues with sprite fonts and things, by all means

**Comment by nielsmh** [comment]
I tested all the new names in-game with both original graphics and OpenGFX sprite fonts, and didn't notice any issues with the characters used.

**Pull request merged**